core/lexer.mll

```diff
@@ -212,6 +213,7 @@ rule lex ctxt nl = parse
   | "||"                                { BARBAR }
   | "&&"                                { AMPAMP }
   | '|'                                 { VBAR }
+  | "|."                                { VBARDOT }
```

Comment:
I don't want to introduce a new token. Let's remove this. I want `.` to be allowed in `evar` position in the parser, if the effect sugar is on.

Reply:
Without this token, the parser doesn't recognize the dot syntax without a space:
```links
links> sig f : () {Ask:() -> a| .}-> a
...... fun f() { do Ask };
f = fun : () {Ask:a|.}-> a

links> sig f : () {Ask:() -> a|.}-> a
***: Parse error: <stdin>:1

  sig f : () {Ask:() -> a|.}-> a
                           ^ 
```

I think it's getting confused with something else, but introducing a token creates the correct behaviour. Unless the problem is actually caused by something else?

Reply:
As discussed, I created a special case for this in the `op` rule, and kept it (at least for now) as a token - this seems to be a clean solution.

Reply:
I have thought a little bit more about this. I think we should keep things simple for now and disallow `|.` and it instead insist you write `| .` for a closed row. I don't envisage closed rows being used at all when the effect sugar is on. I am sort of concerned about this special case, as I think this needs to be replicated elsewhere, in particular when I land the hygienic names patch. So lets remove the token again, and simply the grammar.